lightning_examples/finetuning-scheduler/finetuning-scheduler.py

```diff
@@ -0,0 +1,565 @@
+        mod = import_module("pytorch_lightning.callbacks.finetuning_scheduler")
+        MOCK_HUB_REGISTRY.register_classes(mod, pl.callbacks.Callback)
+    else:
+        raise MisconfigurationException(f"user module key '{key}' not found")
+    registered_list = []
+    # make registered class available by unqualified class name by default
+    if not require_fqn:
+        for n, c in MOCK_HUB_REGISTRY.items():
+            globals()[f"{n}"] = c
+        registered_list = ", ".join([n for n in MOCK_HUB_REGISTRY.names])
+    else:
+        registered_list = ", ".join([c.__module__ + "." + c.__name__ for c in MOCK_HUB_REGISTRY.classes])
+    print(f"Imported and registered the following callbacks: {registered_list}")
+
+
+def instantiate_registered_class(init: Dict[str, Any], args: Optional[Union[Any, Tuple[Any, ...]]] = None) -> Any:
+    """Instantiates a class with the given args and init. Accepts class definitions in the form of a "class_path"
+    or "callback_key" associated with a _Registry.
+
+    Args:
+        init: Dict of the form {"class_path":... or "callback_key":..., "init_args":...}.
+        args: Positional arguments required for instantiation.
+
+    Returns:
+        The instantiated class object.
+    """
+    kwargs = init.get("init_args", {})
+    if args and not isinstance(args, tuple):
+        args = (args,)
+    shortcircuit_local = False if "." in init["class_path"] else True
+    class_module, class_name, args_class = None, None, None
+    if init.get("class_path", None):
+        if not shortcircuit_local:
+            class_module, class_name = init["class_path"].rsplit(".", 1)
+        else:  # class is expected to be locally defined
+            args_class = globals()[init["class_path"]]
+    elif init.get("callback_key", None):
+        callback_path = CALLBACK_REGISTRY.get(init["callback_key"], None) or MOCK_HUB_REGISTRY.get(
+            init["callback_key"], None
+        )
+        assert callback_path, MisconfigurationException(
+            f'specified callback_key {init["callback_key"]} has not been registered'
+        )
+        class_module, class_name = callback_path.__module__, callback_path.__name__
+    else:
+        raise MisconfigurationException(
+            "Neither a class_path nor callback_key were included in a configuration that" "requires one"
+        )
+    if not shortcircuit_local:
+        module = __import__(class_module, fromlist=[class_name])
+        args_class = getattr(module, class_name)
+    return args_class(**kwargs) if not args else args_class(*args, **kwargs)
+
+
+# %%
+# load the pl extension module we want to use. This will import all necessary callbacks.
+module_hub_mock("finetuningscheduler")
+# set notebook-level variables
+AVAIL_GPUS = min(1, torch.cuda.device_count())
+TASK_NUM_LABELS = {"boolq": 2, "rte": 2}
+DEFAULT_TASK = "rte"
+
+# narrow our logging to adapt it for a notebook environment
+for l_key in logging.Logger.manager.loggerDict.keys():
```

Comment:
This might require an extra explanation. IMO, I would remove it, this makes the explication more complex.

Reply:
Good point, code clarity >> shorter logs :smile:  I've removed the log massaging.